src/node_messaging.cc

```diff
@@ -711,6 +711,13 @@ void MessagePort::PostMessage(const FunctionCallbackInfo<Value>& args) {
     return THROW_ERR_MISSING_ARGS(env, "Not enough arguments to "
                                        "MessagePort.postMessage");
   }
+  if (!args[1]->IsNullOrUndefined() && !args[1]->IsObject()) {
+    // Browsers also do not throw on `null` or objects, although it really
+    // should be an array or undefined, thus the mismatch between the checks
+    // above and the actual error message.
+    return THROW_ERR_INVALID_ARG_TYPE(env,
+        "Transfer list argument must be array or missing");
```

Comment:
```suggestion
        "Optional transfer argument must be an array");
```

Reply:
A thought: It might be worthwhile to align the variable names in our docs and error messages with the analogous MDN entry.  https://developer.mozilla.org/en-US/docs/Web/API/Worker/postMessage calls the arguments `message` and `transfer` but our docs et al. use `value` and `transferList` respectively.